feat: Complete user login

CHANMYUNG opened the pull request:
Resolves: #15

CHANMYUNG committed:
chore: Separate spring profiles

CHANMYUNG committed:
feat: Add convertObjectToJsonBytes() method

- Add convertObjectToJsonBytes() method to convert object
  that be used at acceptance test

CHANMYUNG committed:
feat: Add login api & tests

- Add login api with tests

CHANMYUNG committed:
feat: Complete login front-end

- common.js: common functions

Resolves: #15

Lahavas commented on src/main/java/com/woowahan/techcamp/recipehub/user/service/UserService.java:
```diff
@@ -0,0 +1,27 @@
+
+    public User login(LoginDto dto) {
+        System.out.println(dto.getEmail());
+        System.out.println(dto.getPassword());
```
여기 테스트용으로 출력한거 지워주세요~

CHANMYUNG replied:
앗 이런 엄청난 실수를;; 죄송해용

Lahavas commented on src/main/resources/static/js/users/login.js:
```diff
@@ -0,0 +1,88 @@
+    //         ${text}
+    //     </div>
+    //     `;
+    // }
```
위에 주석 삭제해주세요~

ffaass commented on src/main/java/com/woowahan/techcamp/recipehub/user/service/UserService.java:
```diff
@@ -0,0 +1,27 @@
+    private PasswordEncoder passwordEncoder;
+
+    public User login(LoginDto dto) {
+        System.out.println(dto.getEmail());
```
System.out.println 지워주세요

CHANMYUNG replied:
ㅎㅎ 중복입니당

Lahavas commented on src/main/resources/templates/users/login.hbs:
```diff
@@ -0,0 +1,52 @@
+    <script src="/js/users/login.js"></script>
+    <title>Document</title>
+</head>
+<body>
```
base.hbs랑 공통된 코드네요 :)

ryuneeee commented on src/main/java/com/woowahan/techcamp/recipehub/common/security/SessionUtils.java:
```diff
@@ -0,0 +1,17 @@
+import javax.servlet.http.HttpSession;
+
+public class SessionUtils {
+    private static final String USER_SESSION_KEY = "USER_SESSION_KEY";
```
변수이름이랑 값이랑 같은거 노린건가!🤔

CHANMYUNG replied:
맘에 안드시면..무얼로 바꿔드릴까여

Lahavas commented on src/main/resources/templates/users/login.html:
```diff
@@ -0,0 +1,60 @@
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="stylesheet" href="../../static/css/bulma.css">
+    <title>Document</title>
+</head>
```
login.hbs랑 동일한 동작을 하는 파일이네요

CHANMYUNG replied:
헐 이건 올릴 생각 없었는데;; 서버 안띄우고 퍼블리싱하려거 ㅠㅠ 지우겠습니다

ffaass commented on src/main/resources/templates/users/login.html:
```diff
@@ -0,0 +1,60 @@
+</section>
+<div class="notification is-primary">
+    <button class="delete"></button>
+    Primar lorem ipsum dolor sit amet, consectetur
```
이건 뭐죠 ㅋㅋㅋㅋ

CHANMYUNG replied:
ㅠㅠ 이것두요.. 절 죽여주세요..ㅎㅎ

ryuneeee replied:
이거 로렘입숨인데👍

ryuneeee commented on src/main/java/com/woowahan/techcamp/recipehub/user/service/UserService.java:
```diff
@@ -0,0 +1,27 @@
+
+    public User login(LoginDto dto) {
+        System.out.println(dto.getEmail());
+        System.out.println(dto.getPassword());
```
표준 출력 자제 요망~
Logger.debug를 씁시당

ryuneeee commented on src/main/resources/static/js/users/login.js:
```diff
@@ -0,0 +1,88 @@
+        })
+    }
+
+    areLoginFieldsValid() {
```
validateLoginFields 같은게 더 직관적일듯!

ryuneeee replied:
여기서만 쓰는거라면 isValidForm 이라던가...?

CHANMYUNG replied:
헐 이것까지 해결해서 amend할게요 잠시만요 ㅠㅠ

CHANMYUNG commented:
정말 꼼꼼하게 리뷰 잘해주셨네요 감사합니다 👍 

아마추어같은 실수를 한 점은 아마추어니까 봐주세요..ㅋㅋㅋㅋ 절 죽여주시옵소서 ㅠㅠ

ryuneeee reviewed (changes requested):
System.out.println 정도만 바꿔주세용~
나머지는 👍

CHANMYUNG committed:
refactor: Remove useless logs, comments, and so on

Resolves: #15

minzzzy reviewed (approved):
승인

Lahavas commented on src/test/java/com/woowahan/techcamp/recipehub/user/domain/UserTest.java:
```diff
@@ -0,0 +1,20 @@
+import static org.junit.Assert.assertTrue;
+
+public class UserTest {
+    private final String USER_PASSWORD = "USER_PASSWORD";
```
여기 전역 상수가 좋을거같슴다

CHANMYUNG committed:
refactor: Fix 'static' or 'final' to 'static final'

- Experimental results: I could see that junit created test instances
  as many as the number of test cases.
  So final keyword should be made 'static'

Resolves: #15

Lahavas reviewed (approved):
승인이여~
